### PR TITLE
Wgpu0.14to0.16

### PR DESCRIPTION
**Pull request by CaoKha**
[upgrade wgpu to 0.16](https://github.com/CaoKha/yogurt_sim/pull/6)

### CODE REVIEW

File: src/state/texture.rs [hunk 1]
```diff
@@ -52,8 +53,8 @@ impl Texture {
             &rgba,
             wgpu::ImageDataLayout {
                 offset: 0,
-                bytes_per_row: std::num::NonZeroU32::new(4 * dimensions.0),
```
Comment (TrungDinhT) [inline src/state/texture.rs:55]: Why NonZeroU32 at the beginning 😯?

File: src/state/mod.rs [hunk 2]
```diff
@@ -50,13 +51,11 @@ pub struct State {
...
 impl State {
     // Creating some of the wgpu types requires async code
-    pub async fn new(window: &Window) -> Self {
-        let (surface, adapter) = init_surface(window).await;
-
+    pub async fn new(window: Window) -> Self {
         let size = window.inner_size();
-        let config = init_config(size, &surface, &adapter);
-
+        let (surface, adapter, window) = init_surface(window).await;
```
Comment (TrungDinhT) [inline src/state/mod.rs]: Why move `window` in and then return it out while it can be taken by const ref 🤨?

**Review by TrungDinhT — APPROVED** [review]
LGTM! Just some minor questions

File: src/state/mod.rs (hunk 2 above)
Comment (CaoKha) [reply src/state/mod.rs]: fixed. window is now passed by reference

File: src/state/texture.rs (hunk 1 above)
Comment (CaoKha) [reply src/state/texture.rs:55]: Changes from 0.14 to 0.16. `bytes_per_row` type is changed from `NonZeroU32` to `u32`

**Pull request merged**